src/main/java/com/google/sps/model/AuthenticatedHttpServlet.java

```diff
@@ -93,4 +114,52 @@ public void doPost(
       throws IOException {
     response.sendError(400, "POST is not supported");
   }
+
+  /**
+   * Get a Google Credential object from the authentication cookies (idToken & accessToken) in the
+   * HTTP Request
+   *
+   * @param request Http Request sent from client
+   * @return Credential object with accessToken. null if tokens not present / are invalid
+   */
+  private Credential getGoogleCredential(HttpServletRequest request) {
+    Cookie userTokenCookie = ServletUtility.getCookie(request, "idToken");
+    if (userTokenCookie == null) {
+      return null;
+    }
+
+    String idToken = userTokenCookie.getValue();
+
+    if (idToken.isEmpty()) {
+      return null;
+    }
+
+    try {
+      if (!authenticationVerifier.verifyUserToken(userTokenCookie.getValue())) {
+        return null;
+      }
+    } catch (GeneralSecurityException | IOException e) {
+      e.printStackTrace();
+      return null;
+    }
+
+    Cookie accessTokenCookie = ServletUtility.getCookie(request, "accessToken");
+    if (accessTokenCookie == null) {
+      return null;
+    }
+
+    String accessToken = accessTokenCookie.getValue();
+
+    if (accessToken.isEmpty()) {
+      return null;
```

Comment:
I see a lot of `return null;` here, might be worth throwing an exception of some sort.

Reply:
Agreed - I added exceptions for exceptional circumstances (the tokens not being present is not exceptional)